src/framework/mocha/Utilities/MOFunctionArgument.m

```diff
@@ -1095,4 +1095,63 @@ + (NSInteger)structureToJSValue:(JSValueRef *)value inContext:(JSContextRef)ctx
     return c - c0 - 1;
 }
 
++ (BOOL)isHighSierraOrHigher {
+    NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
+    return (version.minorVersion >= 13);
+}
+
+/**
+ This method is only called when running 10.13 or higher.
+ 
+ The method takes a symbol type string like '{CGRect="origin"{CGPoint}"size"{CGSize}}' and creates a dictionary
+ with keys = ["CGPoint", "CGSize"]. Whose values are: ['CGPoint="x"d"y"d', 'CGSize="width"d"height"d']
+ 
+ The method does this by looking up the type descriptions for each of the structs.
+ */
++ (NSDictionary <NSString *, NSString *> *)memberStructs:(NSString *)symbolType {
+    NSMutableDictionary <NSString *, NSString *> *memberStructsDictionary = [NSMutableDictionary new];
+    
+    // First strip off the initial { and final } characters from the symbol type after checking the string is long enough.
+    if (symbolType.length < 3) {
+        return memberStructsDictionary.copy;
+    }
+    symbolType = [symbolType substringWithRange:NSMakeRange(1, symbolType.length - 2)];
+    
+    // Create an array of strings seperated by the { character.
+    NSArray <NSString *> *memberStructs = [symbolType componentsSeparatedByString:@"{"];
+    
+    NSEnumerator <NSString *> *enumerator = [memberStructs objectEnumerator];
+    
+    // Drop the first object as it precedes the first "{" so doesn't represent a struct type.
+    (void)enumerator.nextObject;
+    
+    NSString *structName;
+    while ((structName = enumerator.nextObject)) {
+        NSRange locationOfEndBracket = [structName rangeOfString:@"}"];
+        if (locationOfEndBracket.location == NSNotFound) {
+            continue;
+        }
+        
+        structName = [structName substringWithRange:NSMakeRange(0, locationOfEndBracket.location)];
+        NSString *structNameReplacement = [self structureFullTypeEncodingFromStructureName:structName];
+        if (!structNameReplacement) {
+            continue;
+        }
+        
+        memberStructsDictionary[structName] = [structNameReplacement substringWithRange:NSMakeRange(1, structNameReplacement.length - 2)];
+    }
+    return memberStructsDictionary.copy;
+}
+
++ (NSString *)expandSymbolStructType:(NSString *)symbolType {
+    if (symbolType && self.isHighSierraOrHigher) {
```

Comment:
Where is the information for knowing what type of description we are getting from the bridge support system? What will tell us with greater confidence than the operating system version what type of bridging support we have?

We could attempt to parse as if the description is there for members which are structs and if it isn't because we get back an error then update it then. But then we are dealing with attempting that each time. Which in that case we should go down the cache route.